scipy/integrate/_ivp/radau.py

```diff
@@ -72,6 +72,11 @@ def solve_collocation_system(fun, t, y, h, Z0, scale, tol,
     solve_lu : callable
         Callable which solves a linear system given a LU decomposition. The
         signature is ``solve_lu(LU, b)``.
+    mass_matrix : {None, array_like, sparse_matrix}, optional
+           Defined the constant mass matrix of the system, with shape (n,n).
+           It may be singular, thus defining a problem of the differential-
+           algebraic type (DAE). The default value is None (equivalent to
+           an identity mass matrix).
```

Comment:
Please add an example to the docstring. It's been a few years since I've worked with this sort of thing, so seeing the test reminded me of how this would work.